File: java/67.java
class Solution {
    public String addBinary(String a, String b) {
        List<Integer> l = new ArrayList();
        int i = a.length()-1;
        int j = b.length()-1;
        int carry = 0;
        while(i >= 0 && j >= 0){
            int c = a.charAt(i) - '0';
            int d = b.charAt(j) - '0';
            int x = c + d + carry;
            l.add(x % 2);
            carry = x / 2;
            i --;
            j --;
        }
        while(i >= 0){
            int c = a.charAt(i) - '0';
            i --;
            int x = c + carry;
            l.add(x % 2);
            carry = x / 2;
        }
        while(j >= 0){
            int c = b.charAt(j) - '0';
            j --;
            int x = c + carry;
            l.add(x % 2);
            carry = x / 2;
        }
        if(carry == 1){
            l.add(1);
        }
        StringBuilder s = new StringBuilder();
        for(int o = l.size()-1;o >=0;o--){
            s.append(l.get(o));
        }
        return s.toString();
    }
}

File: go/67.go
func addBinary(a string, b string) string {
	m, n := len(a)-1, len(b)-1
	ans := ""
	carry := 0
	for m >= 0 && n >= 0 {
		c, d := 0, 0
		if a[m] == '1' {
			c = 1
		}
		if b[n] == '1' {
			d = 1
		}
		y := carry + c + d
		carry = y / 2
		x := y % 2
		// fmt.Println(y)
		if x == 1 {
			ans = "1" + ans
		} else {
			ans = "0" + ans
		}
		m--
		n--
	}
	for m >= 0 {
		c := 0
		if a[m] == '1' {
			c = 1
		}
		y := carry + c
		carry = y / 2
		x := y % 2
		if x == 1 {
			ans = "1" + ans
		} else {
			ans = "0" + ans
		}
		m--
	}
	for n >= 0 {
		c := 0
		if b[n] == '1' {
			c = 1
		}
		y := carry + c
		carry = y / 2
		x := y % 2
		if x == 1 {
			ans = "1" + ans
		} else {
			ans = "0" + ans
		}
		n--
	}
	if carry == 1 {
		ans = "1" + ans
	}
	return ans

}